arch/arc/core/cache.c

```diff
@@ -67,34 +67,18 @@ static void dcache_enable(void)
 	dcache_dc_ctrl(DC_CTRL_DC_ENABLE);
 }
 
-
-/**
- *
- * @brief Flush multiple d-cache lines to memory
- *
- * No alignment is required for either <start_addr> or <size>, but since
- * dcache_flush_mlines() iterates on the d-cache lines, a cache line
- * alignment for both is optimal.
- *
- * The d-cache line size is specified either via the CONFIG_CACHE_LINE_SIZE
- * kconfig option or it is detected at runtime.
- *
- * @param start_addr the pointer to start the multi-line flush
- * @param size the number of bytes that are to be flushed
- *
- * @return N/A
- */
-static void dcache_flush_mlines(u32_t start_addr, u32_t size)
+void arch_dcache_flush(void *start_addr, size_t size)
 {
-	u32_t end_addr;
+	uintptr_t end_addr;
 	unsigned int key;
 
 	if (!dcache_available() || (size == 0U)) {
 		return;
 	}
 
-	end_addr = start_addr + size - 1;
-	start_addr &= (u32_t)(~(DCACHE_LINE_SIZE - 1));
+	end_addr = (uintptr_t)start_addr + size;
+	start_addr = (void *)(((uintptr_t)start_addr) &
```

Comment:
consider using `ROUND_DOWN(start_addr, DCACHE_LINE_SIZE)` for clarity. This macro and its companion ROUND_UP are found in include/sys/util.h